department/department_app/views/department_views.py

```diff
@@ -0,0 +1,21 @@
+"""Views department"""
+from rest_framework import generics
+from department_app.rest.serializers import DepartmentListSerializer
+from department_app.models import Department
+
+
+class DepartmentCreateView(generics.CreateAPIView):
+    """Create department"""
+    serializer_class = DepartmentListSerializer
+
+
+class DepartmentListView(generics.ListAPIView):
+    """Read all department"""
+    serializer_class = DepartmentListSerializer
+    queryset = Department.objects.all()
+
+
+class DepartmentDetailView(generics.RetrieveUpdateDestroyAPIView):
+    """Read, update, delete"""
+    serializer_class = DepartmentListSerializer
+    queryset = Department.objects.all()

```

Comment:
Please use ViewSet this will allow to minimize the code.

Reply:
Fixed